File: submissions/sm_017_mukund/week_3/day_6/assignments/pyramidPlay.js
function pyramid(){

    for(i=0;i<4;i++){

        str = ''

        for(j=0;j<9;j++){
            
            if( i == 0 ){

                if( j == Math.floor(9/2) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            if( i == 1 ){

                if( j == (Math.floor(9/2)-1) || j == (Math.floor(9/2)+1) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            if( i == 2 ){

                if( j == Math.floor(9/2) || j == (Math.floor(9/2)-2) || j == (Math.floor(9/2)+2) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            if( i == 3 ){

                if( j == (Math.floor(9/2)+1) || j == (Math.floor(9/2)-1) || j == (Math.floor(9/2)+3) || j == (Math.floor(9/2)-3) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            

        }

        console.log(str)

    }

}

pyramid();
console.log()

function inversePyramid(){

    for(i=0;i<4;i++){

        str = ''

        for(j=0;j<9;j++){
            
            if( i == 3 ){

                if( j == Math.floor(9/2) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            if( i == 2 ){

                if( j == (Math.floor(9/2)-1) || j == (Math.floor(9/2)+1) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            if( i == 1 ){

                if( j == Math.floor(9/2) || j == (Math.floor(9/2)-2) || j == (Math.floor(9/2)+2) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            if( i == 0 ){

                if( j == (Math.floor(9/2)+1) || j == (Math.floor(9/2)-1) || j == (Math.floor(9/2)+3) || j == (Math.floor(9/2)-3) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            

        }

        console.log(str)

    }

}

inversePyramid();
console.log()

function rhombus(){

    for(i=0;i<7;i++){

        str = ''

        for(j=0;j<9;j++){
            
            if( i == 0 || i == 6 ){

                if( j == Math.floor(9/2) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            if( i == 1 || i == 5 ){

                if( j == (Math.floor(9/2)-1) || j == (Math.floor(9/2)+1) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            if( i == 2 || i == 4 ){

                if( j == Math.floor(9/2) || j == (Math.floor(9/2)-2) || j == (Math.floor(9/2)+2) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            if( i == 3 ){

                if( j == (Math.floor(9/2)+1) || j == (Math.floor(9/2)-1) || j == (Math.floor(9/2)+3) || j == (Math.floor(9/2)-3) ){

                    str+=0
    
                }
                else{
    
                    str+='.'
    
                }

            }
            

        }

        console.log(str)

    }

}

rhombus();

File: submissions/sm_105_ashish/week_13/day_5/pyramid_play.py
# making method which will return the layer of pyramid in an array.
def pyramid_arr(level):
    final_arr = []
    for j in range(level):
        arr  = []
        for i in range(level*2+1-2*j): 
            arr.append(".") if i%2 == 0 else arr.append("0")
        final_arr.append(arr)
        arr = []
    final_arr.reverse()
    for k in range(len(final_arr)-1):
        for _ in range(k,len(final_arr)-1):
            final_arr[k].append(".")
            final_arr[k].reverse()
            final_arr[k].append(".")
    return final_arr

# making method to show pyramid using the prviuosly made method:

def pyramid(level):
    pyramid_level = pyramid_arr(level)
    for x in pyramid_level:
        print(''.join(x))
# making pyramid of level 5 
# pyramid(5)

#  making method to show inverse of pyramid using the previousaly made method which give as pyramid level:

def inverse_pyramid(level):
    pyramid_level = pyramid_arr(level)
    pyramid_level.reverse()
    for y in pyramid_level:
        print(''.join(y))

# inverse_pyramid(5)

def rhombus(level):
    pyramid_level = pyramid_arr(level)
    for p in pyramid_level:
        print(''.join(p))
    reverse_pyramid_level = pyramid_arr(level)
    reverse_pyramid_level.reverse()
    for q in range(1,len(reverse_pyramid_level)):
        print(''.join(reverse_pyramid_level[q]))

rhombus(15)
